Directory.Build.targets

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+
+  <Target
+    Name="EnsureJavaScriptCodeUpToDate"
+    BeforeTargets="BeforeBuild"
+    Inputs="$(MSBuildThisFileDirectory)yarn.lock"
+    Outputs="$(RootIntDir)\JsUpToDateCheck\yarn.lock"
+    >
+
+    <Message Text="Checking if Yarn.lock file is 'up to date' by running a 'dry-run' version of yarn install and checking the exit code" />
+    <Exec 
```

Comment:
how much time is this adding to the build?

Reply:
The first time a few seconds. The target runs for each vcx and csproj but since the yarn.lock file copy is in a fixed location this only runs once per build, and as long as yarn.lock doesn't change it won't rerun.

Reply:
a few seconds due to the running of yarn install? I meant more on the ongoing no-op case. MSBuild is particularly bad at spawning new processes, perf-wise (e.g. see autolinking). Can we measure this to get an idea of the tradeoff here?

Reply:
I did not see it taking even a second: since the check is done offline, it feels almost instantaneous.

---
In reply to: [520293380](https://github.com/microsoft/react-native-windows/pull/6474#discussion_r520293380) [](ancestors = 520293380)

Reply:
Note there are as many runs of this target as project files in the solution (this is < 10) I wrote the target to do it's up to date check only on a single file, so the target per solution only runs once taking a little over a sec for me. Otherwise it is a noop, no process launched, just msbuild probing two files and comparing timestamps from metadata.

Reply:
THe number of probes will change if I can get the expression correct to get all package.json files without ones in  node_modules and the path transpilation costs. I do not expect this to be anything significant in the noise of anything else msbuild is doing...

Reply:
I added sample logs in the PR details above.